src/Symfony/Component/Validator/Mapping/Loader/AutoMappingTrait.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Mapping\Loader;
+
+use Symfony\Component\Validator\Constraints\AutoMapping;
+use Symfony\Component\Validator\Mapping\ClassMetadata;
+
+/**
+ * Utility methods to create auto mapping loaders.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+trait AutoMappingTrait
```

Comment:
Should we mark the trait as internal?

Reply:
I don't think so, it helps to create custom loaders, and to prevent inconsistency such as the one introduced by  #33828